### PR TITLE
ARROW-5415: [Release] Release script should update R version everywhere

### PR DESCRIPTION
**Pull request by nealrichardson**
In previous patches, I removed other places where the version was explicitly referenced, making them parse it from `r/DESCRIPTION` where necessary. The only remaining one (added by #4711) is `r/NEWS.md`, and it works a little differently when bumping from a release version to a dev version. So from dev to release, we go from e.g. 

```
# arrow 0.13.0.9000

* Stuff we did in this release
* More stuff
```

to

```
# arrow 0.14.0

* Stuff we did in this release
* More stuff
```

but when bumping after release for the next dev version, it goes to

```
# arrow 0.14.0.9000

# arrow 0.14.0

* Stuff we did in this release
* More stuff
```

so that we have a place to accumulate the list of changes in the next release.

### CODE REVIEW

**Comment by nealrichardson** [comment]
Here's a passing Travis build for the lint/release test job: https://travis-ci.org/nealrichardson/arrow/jobs/551443590

File: dev/release/00-prepare.sh
Comment (kou) [inline dev/release/00-prepare.sh]: Could you use `[ ${type} = "snapshot" ]`  here?

File: dev/release/00-prepare-test.rb
Comment (kou) [inline dev/release/00-prepare-test.rb]: Could you use `#{@release_version}.9000` here?

**Review by kou — COMMENTED** [review]
I'm not sure changes except `dev/release/` and `r/NEWS.md` should be included this "Release script should update R version everywhere" pull request.
(Normally, I don't want to include...)

File: dev/release/00-prepare-test.rb
Comment (nealrichardson) [reply dev/release/00-prepare-test.rb]: Oops, yes, will do.

**Comment by nealrichardson** [comment]
Once #4711 merges I'll push a cleaned up branch; just wanted to make the change visible for review now (and thanks for the review).

**Comment by wesm** [comment]
This can be rebased now

**Comment by nealrichardson** [comment]
Rebased and the lint/release job passed on my Travis: https://travis-ci.org/nealrichardson/arrow/jobs/551551770

**Review by kou — APPROVED** [review]
+1
I've pushed a commit to use `[ ... ]` instead of `[[ ... ]]` for consistency.
We can merge this when CI is green.

**Comment by kou** [comment]
Travis CI: https://travis-ci.org/nealrichardson/arrow/builds/551565121

**Pull request closed without merging**
